### ISSUE TITLE
Unused variable in except clauses differs between py2 and py3 

### ISSUE THREAD

**Issue by moodh** [PyCQA/pyflakes#301]
The following code gives a warning in py2 but not py3.
But only if the try clause is inside a function, otherwise it passes on both.

`test.py:5:25: F841 local variable 'e' is assigned to but never used`

```
def test():
    try:
        raise Exception('foo')
    except Exception as e:
        pass

```

Is this a bug or a new feature? Could not find anything related to this "regression" but currently py3 does not catch issues py2 would have caught. This is a problem when writing compatible code at the very least.

Could be related to this PR: https://github.com/PyCQA/pyflakes/pull/59 (found here https://github.com/PyCQA/pyflakes/blob/master/pyflakes/checker.py#L1325)

**Comment by myint** [comment]
This expected. Python 3 clears the variable while Python 2 does not.

https://stackoverflow.com/questions/29268892/python-3-exception-deletes-variable-in-enclosing-scope-for-unknown-reason#29268974

**Comment by moodh** [comment]
@myint: I understand it's safer than previous versions, but isn't it useful to catch unused code?

**Comment by simonpercivall** [comment]
@myint Apart from being inconsistent; and apart from it _actually_ being unused, it's still IMO a good bug-catching check, because:

```
def test():
    e = ''

    try:
        raise Exception('foo')
    except Exception as e:
        pass

    print(e)
```

_does_ cause an UnboundLocalError in Py3.

**Comment by myint** [comment]
Hmm, I think I read the link to #59, but forgot read the actual example @moodh provided.

**Comment by myint** [comment]
Does #293 fix it for you?

**Comment by moodh** [comment]
Yes, the original bug we found is now caught by your fix, thanks!

**Comment by myint** [comment]
@jdufresne actually fixed this in #293 a month prior. 😄

**Comment by jdufresne** [comment]
Thanks for testing. I updated the PR to reference this issue.
